caboose.go

```diff
@@ -58,6 +58,9 @@ var ErrNotImplemented error = errors.New("not implemented")
 var ErrNoBackend error = errors.New("no available strn backend")
 var ErrBackendFailed error = errors.New("strn backend failed")
 
+// ErrTransient is returned when a transient error(timeouts, not found etc) occurs while fetching from the strn backend.
+var ErrTransient error = errors.New("transient error while fetching from strn backend")
```

Comment:
what is the consumer (e.g. `bifrost-gateway`) supposed to do when it gets this error?
* if this is returned to consumer, i'd expect it to wrap the underlying error rather than replacing it
* if it's meant to be purely internal, it shouldn't be exported